### ISSUE TITLE
Show Wikipedia-style visible notification on pages when Redirect has occured

### ISSUE THREAD

**Issue by JonnyJD** [internetarchive/openlibrary#183]
e.g.
![image](https://github.com/internetarchive/openlibrary/assets/978325/93401962-dfe1-4e22-ab3d-c8e2065ff7d7)

When a redirect is mistakenly added (possibly due to an invalid merge, not exactly sure how it works here) it is difficult to realize and undo this change (at least for somebody else).

I had the problem that http://openlibrary.org/authors/OL13193A was a redirect to http://openlibrary.org/authors/OL575040A (Johann Wolfgang von Goethe).
I didn't do the merge/redirect, but having this author broken didn't seem acceptable to me.

So it took me a while to realize that this actually was a redirect, since on the search
http://openlibrary.org/search/authors?q=Johann+Wolfgang+von+Goethe
the first entry shows the right content in the summary, but the page I arrived at was quite unexpected and had nearly nothing to do with the result I clicked on.

What is missing in my eyes is a feature similar to how it works on wikipedia:
http://en.wikipedia.org/wiki/ISBN
leads to "International Standard Book Number", but it does say  "(Redirected from ISBN)" at the top and when clicking on the "ISBN" link you can go to the original link (without being forwarded again) and make changes there.

Always arriving at http://openlibrary.org/authors/OL575040A I had no clue how to fix OL13193A, which was finally done in [this edit](http://openlibrary.org/authors/OL13193A/Johann_Wolfgang_von_Goethe?b=24&a=23&_compare=Compare&m=diff)

There is probably a trick to append a parameter to the url to stop forwarding and there is probably also an undo link for the original editor. That doesn't help the casual user that wants to fix things though.

**Comment by guyjeangilles** [comment]
@jessamynwest @JonnyJD is this still an issue? I can't spot a problem with the links in the original description.

Assigning @hornc per slack discussions since he applied is tags

**Comment by tfmorris** [comment]
The bad redirect was fixed by the my revert https://openlibrary.org/authors/OL13193A/Johann_Wolfgang_von_Goethe?b=24&a=23&_compare=Compare&m=diff

You can see the previous behavior by going to https://openlibrary.org/authors/OL13193A?v=23. I believe that the proposal that when you arrive at http://openlibrary.org/authors/OL575040A that there be some visual indication that you came via a redirect and have a way to to navigate back (probably to https://openlibrary.org/authors/OL13193A?m=history or something similar for us) is a good one, although perhaps not straightforward to implement for our chains of redirects since it'll go redirect->new->new with slug and by the time we've arrived at the page to be rendered, we may have lost the original URL.

**Comment by mekarpeles** [comment]
Does anyone feel very strongly about this? This mostly `Affects: Librarians`. Maybe this should only show a message if someone `is` a librarian?

**Comment by jessamynwest** [comment]
Yeah I think a message is necessary if you're going to address this but could be only for librarians if they are the only people who see it.

**Comment by tfmorris** [comment]
It sounds like a very useful, but not necessarily high priority, feature to me.

The whole goal of OpenLibrary is that, like Wikipedia, ***everyone*** is a librarian, right? That would imply that it should behave the same way for everyone.

**Comment by mekarpeles** [comment]
I'll leave the call to @cdrini; I'm not sure how necessary this is. I'd probably close the issue, but if others feel it's important, I'm sure it can't hurt to have such a redirect message (if the community still thinks there's clear value).

**Comment by MugillaGurilla** [comment]
@psyren99 I was messing around with this one, but never got around to properly making a PR. I made some changes in `readableurls.py`...

Hopefully it can be of some help :) 

On line 13 ...
```
REDIRECT_MESSAGE = "Redirected from %s"
```


On line 64 
```
notify_on_redirect({
    "path": readable_path,
    "notify_on_same_olid": False
})
```

On line 212, right at the bottom
```
def notify_on_redirect(parameters):
    # user = web.ctx.user
    if (
        get_olid(web.ctx.path) != get_olid(parameters['path']) 
        or parameters['notify_on_same_olid']        
    ):
        add_flash_message('info', REDIRECT_MESSAGE % web.ctx.path)

def get_olid(path):
    return path.split('/')[1]
```

**Comment by RayBB** [comment]
Given the current efforts to have a more focused prioritization and the lack of agreement here I'm closing this for now.

Personally, I think it would be handy but not likely something we want to put energy towards in the near future.

CC: @lokesh to think about as he's working on editing changes.
